smart-contracts/assembly/contracts/NFT/NFT-internals.ts

```diff
@@ -1,282 +1,239 @@
-  // compare the values (bytes) referenced by these pointers.
-  const left = nft1_ownerOf(u64ToBytes(tokenId));
-  return (
-    memory.compare(
-      changetype<usize>(left),
-      changetype<usize>(stringToBytes(address)),
-      left.length,
-    ) == 0
-  );
+function ownerKey(tokenId: u256): StaticArray<u8> {
+  return OWNER_KEY_PREFIX.concat(u256ToBytes(tokenId));
 }
 
 /**
- * @param tokenId - the tokenID
- * @returns true if the token is minted
+ * @param tokenId - the tokenID of the approved token
+ * @returns the key of the allowance in the storage for the given owner and spender
  */
-export function _onlyMinted(tokenId: u64): bool {
-  return Storage.has(ownerTokenKey + tokenId.toString());
+function allowanceKey(tokenId: u256): StaticArray<u8> {
+  return ALLOWANCE_KEY_PREFIX.concat(u256ToBytes(tokenId));
 }
 
 /**
- * Internal function returning the currentSupply
- * @returns u64
+ *
+ * @param owner - the address of the owner
+ * @param operator - the address of the operator
+ * @returns The key of the operator allowance in the storage for the given owner and operator
  */
-export function _currentSupply(): u64 {
-  return bytesToU64(Storage.get(counterKey));
-}
-
-// ==================================================== //
-// ====                 TRANSFER                   ==== //
-// ==================================================== //
-
-export function _transfer(
-  caller: string,
+function operatorAllowanceKey(
   owner: string,
-  recipient: string,
-  tokenId: u64,
-): void {
-  assertIsMinted(tokenId);
-  assertIsOwner(owner, tokenId);
-  assertNotSelfTransfer(owner, recipient);
-  assertIsApproved(owner, caller, tokenId);
-
-  _removeApproval(tokenId);
-  _updateBalanceOf(owner, false);
-  _updateBalanceOf(recipient, true);
-
-  Storage.set(ownerTokenKey + tokenId.toString(), recipient);
+  operator: string,
+): StaticArray<u8> {
+  return OPERATOR_ALLOWANCE_KEY_PREFIX.concat(
+    stringToBytes(owner).concat(stringToBytes(operator)),
+  );
 }
 
-// ==================================================== //
-// ====                 APPROVAL                   ==== //
-// ==================================================== //
-
 /**
- * Store the approved address for a token
+ * Count all NFTs assigned to an owner.
  *
- * @param owner - owner address
- * @param spenderAddress - spender address
- * @param tokenId - The token ID to approve
+ * @param owner - An address for whom to query the balance
  */
-export function _approve(
-  tokenId: u64,
-  owner: string,
-  spenderAddress: string,
-): void {
-  assertIsMinted(tokenId);
-  assertIsOwner(owner, tokenId);
-  assert(!_isApproved(spenderAddress, tokenId), 'Already approved');
-
-  const key = approvedTokenKey + tokenId.toString();
-  Storage.set(key, spenderAddress);
+export function _balanceOf(owner: string): u256 {
+  const key = balanceKey(owner);
+  if (Storage.has(key)) {
+    return bytesToU256(Storage.get(key));
+  }
+  return u256.Zero;
```

Comment:
Any reason for not using ternary operator for such simple instructions?
```suggestion
return Storage.has(key) ? bytesToU256(Storage.get(key)) : u256.Zero;
```